### DISCUSSION TITLE
Does final state block an always eventless transition?

### DISCUSSION THREAD

**Discussion by lostrouter** [statelyai/xstate#3174] [Q&A] [answered]
I couldn't find the answer in the documents.  I have a final state that I want to automatically transition when entered based on a condition of context or event. When I remove the `final` type from `a`, the eventless transition runs, but with `final`, it does not. is this expected? I want this to trigger the `onDone` listener for the interpreter when conditions are not me to transition automatically.
 
```
  a: {
    type: 'final',
    on: {
      REQUEST_PROCCESSED: [{ target: 'x' }],
    },
    always: [{ target: 'b', cond: 'shouldGoToB'}]
  },
  b: {
    type: 'final',
    on: {
      DO_THING:  [{ target: 'c' }]
    }
  },
```

**Comment by davidkpiano** [comment, marked as answer]
A final state should not have transitions; it is final.

Note to self: this should be added to the docs and enforced by TypeScript
